cpp/src/groupby/hash/groupby.cu

```diff
@@ -542,11 +542,7 @@ std::pair<rmm::device_uvector<size_type>, size_type> extract_populated_keys(
 {
   rmm::device_uvector<size_type> populated_keys(num_keys, stream);
 
-  auto get_key = [] __device__(auto const& element) {
-    size_type key, value;
-    thrust::tie(key, value) = element;
-    return key;
-  };
+  auto get_key = [] __device__(auto const& element) { return element.first; };  // first = key
```

Comment:
We can do even better: At line 549 (can't comment on these lines since they were not changed), change to `get_key_iter` and line 440 change to `get_key_iter + map.capacity()`.

```suggestion
  auto get_key_iter = thrust::make_transform_iterator(map.data(), [] __device__(auto const& element) { return element.first; });  
```

Reply:
I originally thought the same thing, but it added an extra line of code. However, I made the change plus another one which I thinks makes it look way nicer / more readable :+1:

Reply:
Agreed. It's more readable in its current form.